server/service/src/boajs/methods/use_graphql.rs

```diff
@@ -0,0 +1,36 @@
+use boa_engine::*;
+use serde::{Deserialize, Serialize};
+use ts_rs::TS;
+use util::constants::SYSTEM_USER_ID;
+
+use crate::{boajs::context::BoaJsContext, boajs::utils::*};
+
+#[derive(Debug, Clone, Serialize, Deserialize, TS)]
+pub(crate) struct UseGraphqlInput {
+    query: String,
+    variables: serde_json::Value,
+}
+
+// TODO this method should accept user id, when call plugin is called it should accept this user id
```

Comment:
There is a bulletpoint for this in carry over

Reply:
If this is running through service, user_id is usually available, I thought in that case we should try to use user_id. I have an example of calling use_graphql from graphql_query plugin, there could be a plugin that does something via user UI action.

Reply:
> Kind of think, either execute via usual graphql API

Might need to chat about this again, passing user credentials to plugin/plugin methods, in service layer is very tricky